### PR TITLE
fix(contact-center): start auto wrap-up after conference exit (CAI-8582)

### PR DESCRIPTION
**Pull request by akulakum**
# COMPLETES https://jira-eng-sjc12.cisco.com/jira/browse/CAI-8582

## This pull request addresses

After some wrap-up entries (including consult/conference), the SDK shows wrap-up UI but never starts `setupAutoWrapupTimer()`, so it never POSTs `/v1/tasks/{id}/wrapup` when the team auto wrap-up interval expires. Native Agent Desktop auto-wraps on `AgentWrapup` for the same interactions. Normal end and consult-transfer already work because those payloads already carry `wrapUpRequired` into `updateTaskData`.

Root cause: several wrap-up ingresses wrote raw task data without `wrapUpRequired`. The existing timer only starts when `this.data.wrapUpRequired === true` (and the team profile has auto wrap-up enabled). A later `AgentWrapup` mapped `TASK_WRAPUP` with the flag on the state-machine event only; `updateTaskData` used the raw payload, and `WRAPPING_UP` ignored `TASK_WRAPUP`.

## by making the following changes

- Stamp `wrapUpRequired` on wrap-up ingress payloads so `updateTaskData` / `setupAutoWrapupTimer` see the same flag the state machine already uses to enter `WRAPPING_UP`:
  - `AGENT_WRAPUP` → `true` on the payload written by TaskManager (mapper already did this)
  - `AGENT_CONSULT_CONFERENCE_ENDED`, `PARTICIPANT_LEFT_CONFERENCE` (self left only), `AGENT_CONFERENCE_TRANSFERRED`
  - HTTP `exitConference` → `EXIT_CONFERENCE_SUCCESS` (`Voice.ts`; this path never goes through TaskManager)
- Stamp only from existing wrap-up signals (`wrapUpRequired`, `agentsPendingWrapUp`, `isWrapUp`, `shouldWrapUpForThisAgent` / this agent left). Remaining agents still in a conference stay `false`.
- Handle late `TASK_WRAPUP` while already `WRAPPING_UP` with existing `updateTaskData` + `emitTaskWrapup` (stay in state).
- Auto-wrap POST remains profile-gated in `Task.setupAutoWrapupTimer()` (`autoWrapup !== false`, valid interval, default reason). No new timer or wrap-up API.
- Spec-currency: `WRAPPING_UP` accepts `TASK_WRAPUP` (stay + update/emit).

### Change Type

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios were tested

Automated (`yarn workspace @webex/contact-center test:unit`):
- TaskManager mapper/payload: `AGENT_WRAPUP`, `AGENT_CONSULT_CONFERENCE_ENDED`, `PARTICIPANT_LEFT_CONFERENCE` (self left vs other left), `AGENT_CONFERENCE_TRANSFERRED` — `wrapUpRequired` set only for the wrapping agent.
- Voice `exitConference`: `EXIT_CONFERENCE_SUCCESS` `taskData` includes `wrapUpRequired` when this agent should wrap.
- State machine: wrap-up entry with stamped flag → `WRAPPING_UP` + `emitTaskWrapup`; late `TASK_WRAPUP` stays `WRAPPING_UP`. Remaining-agent `PARTICIPANT_LEAVE` does not wrap.
- `Task.setupAutoWrapupTimer`: profile `autoWrapup: true` starts the timer; `autoWrapup: false` does not.

Manual (recommended against this SDK build):
- Normal inbound end still auto-wraps when the team has auto wrap-up enabled.
- Consult-then-transfer still auto-wraps.
- Consult/conference, then the wrapping agent leaves: wrap-up UI, countdown, `POST /v1/tasks/{id}/wrapup` at the team interval; remaining agent stays on the call.
- Same conference path with auto wrap-up disabled: wrap-up UI, no timer, no POST, manual wrap still works.

## The GAI Coding Policy And Copyright Annotation Best Practices ##
 

 
- [ ] GAI was not used (or, no additional notation is required)
- [ ] Code was generated entirely by GAI
- [x] GAI was used to create a draft that was subsequently customized or modified
- [ ] Coder created a draft manually that was non-substantively modified by GAI (e.g., refactoring was performed by GAI on manually written code)
- [x] Tool used for AI assistance (GitHub Copilot / Other - specify)
  - [ ] Github Copilot
  - [x] Other - Please Specify: Cursor
- [x] This PR is related to
  - [ ] Feature
  - [x] Defect fix
  - [ ] Tech Debt
  - [ ] Automation

### I certified that

- [x] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [x] I discussed changes with code owners prior to submitting this pull request
- [x] I have not skipped any automated checks
- [x] All existing and new tests passed
- [x] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

File: packages/@webex/contact-center/test/unit/spec/services/task/Task.ts [hunk 1]
```diff
@@ -250,6 +256,33 @@ describe('Task (base class)', () => {
...
+    };
+    const enabled = new DummyTask(
+      dummyContact,
+      createTaskData({wrapUpRequired: false}) as TaskData,
+      {wrapUpProps},
+      'agent-1'
+    );
+
+    expect(enabled.autoWrapup).toBeUndefined();
+    enabled.updateTaskData(createTaskData({wrapUpRequired: true}) as TaskData);
```
[the remark this answers is not in the document]
Comment (akulakum) [reply packages/@webex/contact-center/test/unit/spec/services/task/Task.ts]: Addressed. The profile-gate test now wraps `updateTaskData` in `try/finally` and calls `enabled.cancelAutoWrapupTimer()` so the live 25s `AutoWrapup` timeout cannot leak or fire `wrapup()` after teardown.

File: packages/@webex/contact-center/src/services/task/TaskManager.ts [hunk 2]
```diff
@@ -195,6 +203,94 @@ export default class TaskManager extends EventEmitter {
...
+    if (payload.agentsPendingWrapUp?.includes(agentId)) {
+      return true;
+    }
+
+    return payload.interaction?.participants?.[agentId]?.isWrapUp === true;
+  }
+
+  /**
+   * Stamp wrapUpRequired on conference-exit / wrap-up ingress so updateTaskData
+   * and the state machine see the same flag. Remaining agents stay false.
```
[the remark this answers is not in the document]
Comment (akulakum) [reply packages/@webex/contact-center/src/services/task/TaskManager.ts]: Addressed. Documented the wrap-up ingress stamp in the owning artifacts:
- `task-spec.md` — `TASK-R-011` plus a note that auto wrap-up starts only when ingress data has `wrapUpRequired === true`
- `ai-docs/CONTRACTS.md` — `task.events` now covers `task:wrapup` / `task.autoWrapup` on stamped ingresses and late `AgentWrapup`
- `.sdd/manifest.json` — `src/services/task` `provides` includes the TaskManager/Voice stamp so profile-gated auto wrap-up can start

No new event constant. Remaining conference agents stay unstamped.

**Comment by rsarika** [comment]
A few important follow-ups before merge:

1. Preserve agentsPendingWrapUp precedence in every branch. The conference-transfer path currently effectively does explicit signals OR consultingAgentId === agentId, which can mark this agent for wrap-up even when a non-empty pending list explicitly excludes them. When the pending list is non-empty, return only pending.includes(agentId); apply initiator fallback only when the list is empty or absent.

2. wrapUpGuardContext is probably unnecessary. It constructs a partial/fake TaskContext and couples TaskManager to state-machine guard policy. Prefer a small pure helper for self-departure that accepts previous task data, current payload, agent ID, and participant ID. This also matters because the current context uses the new payload as both previous and current state, so the guard cannot detect the “participant entry was removed” representation.

3. Normalize/stamp wrapUpRequired once in prepareEventContext, before the generic updateTaskData and state-machine mapping. The current implementation stamps in both ingress preparation and individual mapper cases, which duplicates the decision logic.

4. Add one end-to-end timer test for conference exit: verify task.data.wrapUpRequired === true, autoWrapup is armed, fake timers advance, and contact.wrapup() is called. The existing assertions mainly verify the stamped state-machine payload, not that auto-wrapup actually executes.

The overall ingress-stamping approach is the right direction; these changes keep the PR focused and protect the remaining-agent scenarios.

**Comment by rsarika** [comment]
PR-scoped follow-up for TaskManager.ts:

Within this PR diff, I recommend:

- Remove wrapUpGuardContext and the TaskContext/guards/shouldWrapUpForThisAgent imports if they are replaced by a small pure helper that accepts previous task data, current payload, agent ID, and participant ID.
- Stamp wrapUpRequired once in prepareEventContext() and remove the duplicate stamping from mapEventToTaskStateMachineEvent().
- Preserve agentsPendingWrapUp precedence in the conference-transfer branch; an initiator fallback must not override a non-empty list that excludes the current agent.
- Avoid explicitSignals || shouldWrapUpForThisAgent() for participant leave, because it reintroduces the owner/isConsulted fallback.
- Add one timer-execution assertion for conference exit: after the stamped ingress, advance fake timers and verify contact.wrapup() is called.

These comments apply only to the changes introduced by this PR; no broader cleanup of TaskManager.ts is requested.

**Review by rsarika — APPROVED** [review]

**Comment by akulakum** [comment]
@rsarika Addressed in the latest commit — mapped to your five bullets:

1. **Removed `wrapUpGuardContext`** and the `TaskContext` / `guards` / `shouldWrapUpForThisAgent` imports. Leave detection is a local helper: `participantId === agentId` or `agentId` missing from the current `payload.participants` (EP-DN). We did **not** add a previous-vs-current snapshot API — the existing leave guard already uses the current payload only; the fake `TaskContext` was only supplying `agentId`.

2. **`wrapUpRequired` is stamped once** in `prepareEventContext()` before `updateTaskData` / mapper. Mapper cases pass `taskData: payload` through unchanged.

3. **Pending list wins on conference transfer.** If `agentsPendingWrapUp` is nonempty, the stamp is only `includes(self)`. `consultingAgentId` and `transferConferenceRequested` apply only when the list is empty or absent.

4. **Dropped `explicitSignals || shouldWrapUpForThisAgent()` on leave** so owner / `isConsulted` are not reintroduced. Replacement: remaining agent → explicit signals only; left + nonempty pending → `includes(self)`; left + empty/absent pending → `true` (Agent-1 still auto-wraps when the backend omits the pending list).

5. **Added a timer-execution test:** stamped `wrapUpRequired` + profile `autoWrapup` → advance fake timers → `contact.wrapup()` is called; timer cancelled/restored.

TASK-R-011 updated to match. No `shouldWrapUp` guard-logic change and no broader TaskManager cleanup.

File: packages/@webex/contact-center/src/services/task/TaskManager.ts [hunk 3]
```diff
@@ -195,6 +202,129 @@ export default class TaskManager extends EventEmitter {
...
+    return Array.isArray(payload.agentsPendingWrapUp) && payload.agentsPendingWrapUp.length > 0;
+  }
+
+  /**
+   * Transfer initiator fallback: widgets set transferConferenceRequested on the
+   * actor; desktop success payloads may include consultingAgentId. Only used
+   * when agentsPendingWrapUp is empty or absent.
+   */
+  private static isSelfConferenceTransferInitiator(
+    payload: WebSocketPayload,
```
Comment (rsarika) [inline packages/@webex/contact-center/src/services/task/TaskManager.ts:275]: [P1] Do not infer self-leave from a partial participant map when another participant is explicitly named

When `participantId` is another agent and the payload has a partial `interaction.participants` map that omits the current agent, this returns `true`. `TaskManager` updates the task before sending the state-machine event, so it can set `wrapUpRequired` and arm the auto-wrapup timer for the remaining agent even though the state-machine guard later keeps that agent out of `WRAPPING_UP`. First honor `participantId !== agentId` as a negative result; then only use the missing-participant fallback with the existing previous-task/EP-DN evidence. Please add a regression test asserting the remaining agent has `wrapUpRequired === false` and no auto-wrapup timer.

**Review by rsarika — COMMENTED** [review]
Blocking review comment for the participant-leave edge case.

File: packages/@webex/contact-center/src/services/task/TaskManager.ts (hunk 3 above)
Comment (akulakum) [reply packages/@webex/contact-center/src/services/task/TaskManager.ts:275]: Addressed. Named other `participantId` is never self-leave — we do not inspect a partial `participants` map in that case.

Missing-map fallback is EP-DN only (`participantId` absent). When a previous task map exists, it must have contained self. Remaining-agent regression: `wrapUpRequired === false` and `autoWrapup` stays undefined.

Stamp-only; `didCurrentAgentLeaveMainInteraction` / `shouldWrapUp` unchanged.

File: packages/@webex/contact-center/src/services/task/TaskManager.ts [hunk 4]
```diff
@@ -195,6 +202,129 @@ export default class TaskManager extends EventEmitter {
...
+        };
+      }
+      if (TaskManager.hasNonemptyPendingWrapUp(payload)) {
+        return {
+          ...payload,
+          wrapUpRequired: Boolean(agentId && payload.agentsPendingWrapUp?.includes(agentId)),
+        };
+      }
+
+      return {...payload, wrapUpRequired: true};
```
[the remark this answers is not in the document]
Comment (akulakum) [reply packages/@webex/contact-center/src/services/task/TaskManager.ts]: Addressed. After a real self-leave, empty/absent `agentsPendingWrapUp` no longer hard-stamps `wrapUpRequired: true`.

Consulted non-owner (`isConsulted === true` on the payload or `task.data`, and owner !== self) stays `false` unless an explicit signal includes this agent — same outcome as Voice `exitConference` / `shouldWrapUpForThisAgent`. We did not re-import that guard into TaskManager.

Agent-1 / owner / not-consulted leave with empty pending still stamps `true` (CAI-8582).

File: packages/@webex/contact-center/src/services/task/TaskManager.ts [hunk 5]
```diff
@@ -645,6 +816,15 @@ export default class TaskManager extends EventEmitter {
       adjustedPayload = TaskManager.preserveConfirmedOwner(task, adjustedPayload);
     }
 
+    if (WRAP_UP_STAMP_EVENTS.has(eventType)) {
+      adjustedPayload = TaskManager.stampWrapUpRequiredForEvent(
+        eventType,
+        adjustedPayload,
+        this.agentId,
+        task
+      );
```
[the remark this answers is not in the document]
Comment (akulakum) [reply packages/@webex/contact-center/src/services/task/TaskManager.ts:838]: Addressed. We gated the leave stamp instead of adding a `PARTICIPANT_LEAVE` transition on `HOLD_INITIATING` (existing SM tests require that event to be ignored).

On `PARTICIPANT_LEFT_CONFERENCE`, if the actor snapshot is present and is not `HELD` / `RESUME_INITIATING` / `CONSULTING` / `CONSULT_INITIATING` / `CONFERENCING`, the payload is returned unchanged — so `CONNECTED`, `HOLD_INITIATING`, and `CONF_INITIATING` no longer get `wrapUpRequired` stamped or an auto-wrap timer. No snapshot / no task still uses the existing leave-stamp rules.

`AGENT_WRAPUP`, conference-end, and transfer stamps are unchanged.

File: packages/@webex/contact-center/src/services/task/state-machine/TaskStateMachine.ts [hunk 6]
```diff
@@ -1249,6 +1249,10 @@ export function getTaskStateMachineConfig(uiControlConfig: UIControlConfig) {
         // Only emit wrapup event on entry - task:end should only be emitted when COMPLETED
         entry: ['emitTaskWrapup'],
         on: {
+          // Late AgentWrapup after conference exit already entered WRAPPING_UP
+          [TaskEvent.TASK_WRAPUP]: {
+            actions: ['updateTaskData', 'emitTaskWrapup'],
```
[the remark this answers is not in the document]
Comment (akulakum) [reply packages/@webex/contact-center/src/services/task/state-machine/TaskStateMachine.ts]: Addressed. Late `TASK_WRAPUP` while already in `WRAPPING_UP` always runs `updateTaskData`. `emitTaskWrapup` runs only when `wrapUpRequired` was not already `true` (checked before the update).

Stamped leave/end/transfer that already published `task:wrapup` no longer emits again. Ownership entry without `wrapUpRequired` still emits when late `AgentWrapup` applies the flag. The skip is only on this handler — not inside `Task.emitTaskWrapup` globally.

File: packages/@webex/contact-center/src/services/task/TaskManager.ts [hunk 7]
```diff
@@ -195,6 +210,175 @@ export default class TaskManager extends EventEmitter {
...
+          ...payload,
+          wrapUpRequired: Boolean(agentId && payload.agentsPendingWrapUp?.includes(agentId)),
+        };
+      }
+
+      return {
+        ...payload,
+        wrapUpRequired:
+          TaskManager.wrapUpRequiredFromExplicitSignals(payload, agentId) ||
+          !TaskManager.isConsultedNonOwner(payload, agentId, task),
```
[the remark this answers is not in the document]
Comment (akulakum) [reply packages/@webex/contact-center/src/services/task/TaskManager.ts:362]: Addressed. The leave stamp is unchanged. Incoming `WRAPPING_UP` transitions no longer call `emitTaskWrapup`; the state's entry action is the sole publisher, so consumers get one `task:wrapup`.

Hydrate already relied on entry. Late `TASK_WRAPUP` while already wrapping still uses `applyLateAgentWrapup` (data update always; emit only if wrap-up was not already published). No change to `Task.emitTaskWrapup` or Voice.

File: packages/@webex/contact-center/src/services/task/voice/Voice.ts [hunk 8]
```diff
@@ -1426,9 +1427,23 @@ export default class Voice extends Task implements IVoice {
...
+        const responseTaskData = (response?.data ?? {}) as TaskData;
+        const snapshot = this.stateMachineService.getSnapshot();
+        const wrapUpContext = (snapshot?.context ?? {
+          uiControlConfig: this.uiControlConfig,
+          taskData: this.data,
+        }) as TaskContext;
+        const wrapUpRequired = shouldWrapUpForThisAgent(wrapUpContext, {
+          ...this.data,
+          ...responseTaskData,
+        });
```
[the remark this answers is not in the document]
Comment (akulakum) [reply packages/@webex/contact-center/src/services/task/voice/Voice.ts]: Addressed. Voice `exitConference` still uses `shouldWrapUpForThisAgent` first. If that is false and `agentsPendingWrapUp` is empty/absent, we apply the same self-exit fallback as TaskManager leave: `true` unless this agent is a consulted non-owner (`isConsulted === true` and owner !== self).

Owner-changed HTTP success that omits `isConsulted` now stamps `wrapUpRequired: true`. Consulted non-owner stays `false`. We did not change `shouldWrapUp` / `shouldWrapUpForThisAgent` (those still run for every agent on conference-end).

File: packages/@webex/contact-center/src/services/task/voice/Voice.ts [hunk 9]
```diff
@@ -1426,9 +1445,28 @@ export default class Voice extends Task implements IVoice {
...
+          taskData: this.data,
+        }) as TaskContext;
+        const mergedTaskData = {
+          ...this.data,
+          ...responseTaskData,
+        };
+        let wrapUpRequired = shouldWrapUpForThisAgent(wrapUpContext, mergedTaskData);
+        // Confirmed self-exit: empty/absent pending matches TaskManager leave stamp.
+        if (!wrapUpRequired && !hasNonemptyPendingWrapUp(mergedTaskData)) {
+          wrapUpRequired = !isConsultedNonOwner(mergedTaskData, this.data);
```
[the remark this answers is not in the document]
Comment (akulakum) [reply packages/@webex/contact-center/src/services/task/voice/Voice.ts]: Addressed. Pending-list precedence on Voice HTTP exit uses the correlated response only.

If the success payload omits `agentsPendingWrapUp`, we drop any cached list before `shouldWrapUpForThisAgent` and the self-exit fallback. A nonempty **response** list still wins. Stale other-agent pending on `this.data` no longer stamps `false`.

No change to `shouldWrapUp` / TaskManager.

**Review by bhabalan — APPROVED** [review]

File: packages/@webex/contact-center/src/services/task/voice/Voice.ts [hunk 10]
```diff
@@ -1426,9 +1445,32 @@ export default class Voice extends Task implements IVoice {
...
+        }
+
         this.stateMachineService.send({
           type: TaskEvent.EXIT_CONFERENCE_SUCCESS,
-          taskData: response.data,
+          taskData: {
+            ...responseTaskData,
+            wrapUpRequired,
+          },
```
[the remark this answers is not in the document]
Comment (akulakum) [reply packages/@webex/contact-center/src/services/task/voice/Voice.ts:1473]: Addressed. Added shared `exitConferenceSuccessTransitions()` and wired `EXIT_CONFERENCE_SUCCESS` on `CONNECTED`, `HELD`, and `CONFERENCING` so Voice HTTP exit matches the states where `exitConference()` is allowed (including ownership-transfer while the actor stays `CONNECTED`).

Regression: TaskStateMachine CONNECTED + conference-in-progress task data → `WRAPPING_UP`; Voice `exitConference()` E2E snapshot → `WRAPPING_UP`. TASK-R-011 / SM spec updated.

File: packages/@webex/contact-center/src/services/task/state-machine/TaskStateMachine.ts [hunk 11]
```diff
@@ -1249,6 +1248,10 @@ export function getTaskStateMachineConfig(uiControlConfig: UIControlConfig) {
...
```
[the remark this answers is not in the document]
Comment (akulakum) [reply packages/@webex/contact-center/src/services/task/state-machine/TaskStateMachine.ts]: Addressed without `enqueueActions`. `WRAPPING_UP.on.TASK_WRAPUP` is now a guarded pair: if `context.taskData.wrapUpRequired === true` before the event, only `updateTaskData`; otherwise `updateTaskData` + `emitTaskWrapup`.

Tests cover stamped exit then late `AgentWrapup` (no extra emit) and entry without `wrapUpRequired` then late event (emit once). Ingress stamping and leave entry-only emit unchanged.

File: packages/@webex/contact-center/src/services/task/TaskManager.ts [hunk 12]
```diff
@@ -195,6 +210,175 @@ export default class TaskManager extends EventEmitter {
...
+    payload: WebSocketPayload,
+    agentId?: string,
+    task?: ITask
+  ): WebSocketPayload {
+    if (!WRAP_UP_STAMP_EVENTS.has(eventType)) {
+      return payload;
+    }
+
+    if (eventType === CC_EVENTS.AGENT_WRAPUP) {
+      return {...payload, wrapUpRequired: true};
```
[the remark this answers is not in the document]
Comment (akulakum) [reply packages/@webex/contact-center/src/services/task/TaskManager.ts:328]: Addressed. Added `TASK_WRAPUP` → `WRAPPING_UP` on `HOLD_INITIATING`, `RESUME_INITIATING`, `CONSULT_INITIATING`, and `CONF_INITIATING` with `updateTaskData` + `markEnded` only (no transition emit; entry publishes once).

Test: `enters WRAPPING_UP on TASK_WRAPUP while HOLD_INITIATING and emits wrap-up once`. `AGENT_WRAPUP` ingress stamp unchanged; with these routes, `updateTaskData` + SM transition stay aligned so auto-wrap and `WRAPUP_COMPLETE` are not stuck in a transient state. CAI-8582 conference-exit paths unchanged.

File: packages/@webex/contact-center/src/services/task/state-machine/TaskStateMachine.ts [hunk 13]
```diff
@@ -90,6 +89,18 @@ export function getTaskStateMachineConfig(uiControlConfig: UIControlConfig) {
     },
   ];
 
+  const exitConferenceSuccessTransitions = () => [
+    {
+      guard: guards.shouldWrapUp,
+      target: TaskState.WRAPPING_UP,
+      actions: ['updateTaskData', 'markEnded', 'clearConsultState', 'emitTaskWrapup'],
```
[the remark this answers is not in the document]
Comment (akulakum) [reply packages/@webex/contact-center/src/services/task/state-machine/TaskStateMachine.ts]: Addressed. Removed `emitTaskWrapup` from the wrap-up branch of shared `exitConferenceSuccessTransitions()`; `WRAPPING_UP` entry is the sole publisher (same policy as `PARTICIPANT_LEAVE` → wrap-up).

Test: `emits wrap-up once on EXIT_CONFERENCE_SUCCESS while CONNECTED`. TASK-R-011 / SM spec updated. Auto-wrap timer unchanged (`updateTaskData` still runs on the transition).

File: packages/@webex/contact-center/src/services/task/state-machine/ai-docs/task-state-machine-spec.md [hunk 14]
```diff
@@ -1092,6 +1098,12 @@ It is instantiated by `Task` and receives mapped backend/user events through `se
 
 - Actions: `updateTaskData`, `markEnded`, `emitTaskWrapup`
 
+- `EXIT_CONFERENCE_SUCCESS` -> `WRAPPING_UP` or `TERMINATED`
+
+- Guard: `guards.shouldWrapUp` or default
+
+- Actions: `updateTaskData`, `markEnded`, `clearConsultState`, `emitTaskWrapup` or `emitTaskEnd`
```
[the remark this answers is not in the document]
Comment (akulakum) [reply packages/@webex/contact-center/src/services/task/state-machine/ai-docs/task-state-machine-spec.md]: Addressed. **HELD** `EXIT_CONFERENCE_SUCCESS` now matches **CONNECTED** / **CONFERENCING**: transition actions are `updateTaskData`, `markEnded`, `clearConsultState` (or `emitTaskEnd` on terminate); `WRAPPING_UP` entry emits `task:wrapup`. Also aligned **CONNECTED/HELD** `TASK_WRAPUP` and `TRANSFER_CONFERENCE_SUCCESS` → `WRAPPING_UP` action lists in `task-state-machine-spec.md` and `ARCHITECTURE.md` (CONSULTING transfer wrap branch).

File: packages/@webex/contact-center/src/services/task/TaskManager.ts [hunk 15]
```diff
@@ -195,6 +210,175 @@ export default class TaskManager extends EventEmitter {
...
+        ...payload,
+        wrapUpRequired: TaskManager.wrapUpRequiredFromExplicitSignals(payload, agentId),
+      };
+    }
+
+    return {
+      ...payload,
+      wrapUpRequired:
+        TaskManager.wrapUpRequiredFromExplicitSignals(payload, agentId) ||
+        TaskManager.isSelfConferenceTransferInitiator(payload, agentId, task),
```
[the remark this answers is not in the document]
Comment (akulakum) [reply packages/@webex/contact-center/src/services/task/TaskManager.ts:378]: Addressed. TaskManager initiator fallback stamping (374–378) is unchanged — that arms auto-wrap when widgets set `transferConferenceRequested` without payload signals.

Removed `emitTaskWrapup` from the `shouldWrapUp` branches of `TRANSFER_CONFERENCE_SUCCESS` in **CONSULTING** and **CONFERENCING**; `WRAPPING_UP` entry is the sole publisher (same as participant-leave and exit-success).

Tests: `emits wrap-up once on TRANSFER_CONFERENCE_SUCCESS from CONSULTING/CONFERENCING when wrap-up is required`. Spec updated for transfer wrap branches.

File: packages/@webex/contact-center/src/services/task/TaskManager.ts [hunk 16]
```diff
@@ -195,6 +210,175 @@ export default class TaskManager extends EventEmitter {
...
+    }
+
+    if (eventType === CC_EVENTS.AGENT_WRAPUP) {
+      return {...payload, wrapUpRequired: true};
+    }
+
+    if (eventType === CC_EVENTS.AGENT_CONSULT_CONFERENCE_ENDED) {
+      return {
+        ...payload,
+        wrapUpRequired: TaskManager.wrapUpRequiredFromExplicitSignals(payload, agentId),
```
[the remark this answers is not in the document]
Comment (akulakum) [reply packages/@webex/contact-center/src/services/task/TaskManager.ts:334]: Addressed. emitTaskWrapup on CONFERENCE_END → WRAPPING_UP; entry is the sole publisher (same as leave, exit-success, transfer). Test: emits wrap-up once on CONFERENCE_END when wrap-up is required. Spec updated. TaskManager stamp unchanged.

File: packages/@webex/contact-center/src/services/task/TaskManager.ts (hunk 12 above)
[the remark this answers is not in the document]
Comment (akulakum) [reply packages/@webex/contact-center/src/services/task/TaskManager.ts:328]: Removed transition emitTaskWrapup on TASK_WRAPUP → WRAPPING_UP for OFFERED (shouldWrapUp), CONSULTING, and CONFERENCING; entry remains sole publisher. Three emit-once unit tests added; spec updated. OUTBOUND_FAILED / CONTACT_ENDED transition emits unchanged (follow-up).
